libsolidity/analysis/TypeChecker.cpp

```diff
@@ -2039,6 +2039,9 @@ void TypeChecker::typeCheckFunctionGeneralChecks(
 	solAssert(!!_functionType, "");
 	solAssert(_functionType->kind() != FunctionType::Kind::ABIDecode, "");
 
+	if (_functionType->kind() == FunctionType::Kind::ArrayPop)
```

Comment:
I guess something like this would be also needed for yul. Not sure whether it should be done differently.

Reply:
Not sure if returning early is the way to proceed here. What goes wrong here?

Reply:
The problem is that, because I used the `FunctionType` to store the actual array type within the parameters, that the type-checker wants to check for the supplied arguments. Thats why I just "ignore" the check here, if `FunctionType::Kind::ArrayPop` is used. But I'm not sure whether we want to do it like this. Seem to be not a very nice solution.